packages/cmd/src/services/cache/set.ts

```diff
@@ -82,6 +85,35 @@ export async function handleSetCache() {
   });
 
   success('Cache uploaded. Cache ID: %s', result.cacheId);
+
+  if (saveToHistory) {
+    if (uploadPaths.length > 0) {
+      await handleArchiveUpload({
```

Comment:
Let's do the two archive uploads simultaneously, so we don't extend the total CI time by much for this feature